### ISSUE TITLE
🚧Fix nuget package check issue

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/CICD#129]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Fix an issue with the NuGet package checking process not working correctly.  Investigation of a different issue revealed that the package name used for checking NuGet package versions is hard coded to a value of _**CICDTest**_.

This means that proper checking of a NuGet package of a particular version is returning false positives.

**Issue Location:**
File: CICD.Requirements
Method: `ThatTheNugetPackageDoesNotExist()`
Line #: 1619

```cs
try
{
    packageVersions = nugetService.GetNugetVersions("CICDTest").Result; 👈🏼 //Issue is here
}
catch (AggregateException e) when (e.InnerException is HttpRequestException exception)
{
    if (exception.StatusCode == HttpStatusCode.NotFound)
    {
        return true;
    }
}
```

Also, pull the package ID from the project file.  First, check if this capability is part of NUKE.  If it is not, build a service that can perform this function. Once this is done, use this capability to pull the NuGet package ID from the project file and use that value as the input into the `NugetDataService`.

### **NuGet Version Checking Process:**
Change the way that the `NugetDataService` pulls package version information.

For NuGet search API documentation, go [here](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource)

Use this to use the new API URL instead to get information about a package.  The returned JSON data models will have to be created and then using LINQ, return just a string array of versions for comparison purposes.

**Things to consider:**
- If the query JSON result returns a data JSON array that is empty, this means that the package was not found at all.  In this case, the package does not exist and can be published.
- The results could return multiple packages that are not wanted.  The filtering of the results will have to be further reduced by checking the _**id**_ value of the JSON object.  This way there is no chance of false positives.

### Acceptance Criteria

- [x] Fix issue with hardcode package name for NuGet checks
- [x] Get the package name from the project file before using it.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.
